code/modules/mob/living/simple_animal/hostile/terror_spiders/procs.dm

```diff
@@ -0,0 +1,447 @@
+	return 0
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/msg_terrorspiders(var/msgtext)
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
+		if (T.health > 0)
+			to_chat(T, "<span class='alien'>TerrorSense: " + msgtext + "</span>")
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/CountSpiders()
+	var/numspiders = 0
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
+		if (T.health > 0 && !T.spider_placed && spider_awaymission == T.spider_awaymission)
+			numspiders += 1
+	return numspiders
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/CountAllSpiders()
+	var/numspiders = 0
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
+		numspiders += 1
+	return numspiders
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/CreatePath(var/mygoal)
+	var/m2 = get_turf(src)
+	if (m2 == mylocation)
+		chasecycles++
+		ClearObstacle(get_turf(mygoal))
+		if (chasecycles >= 3)
+			chasecycles = 0
+			if (spider_opens_doors)
+				var/tgt_dir = get_dir(src,mygoal)
+				for(var/obj/machinery/door/airlock/A in view(1,src))
+					if (A.density)
+						try_open_airlock(A)
+				for(var/obj/machinery/door/firedoor/F in view(1,src))
+					if (tgt_dir == get_dir(src,F) && F.density && !F.blocked)
+						visible_message("<span class='danger'>\The [src] pries open the firedoor!</span>")
+						F.open()
+
+	else
+		mylocation = m2
+		chasecycles = 0
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/ClearObstacle(var/turf/target_turf)
+	//Old, easy code.
+	//DestroySurroundings()
+
+	// *****BUG***** This does not allow spiders to smash windoors (e.g: UO71 bar windoor) for some reason.
+	// New, adapted from DestroySurroundings, let's see if it works.
+
+	var/list/valid_obstacles = list(/obj/structure/window, /obj/structure/closet, /obj/structure/table, /obj/structure/grille, /obj/structure/rack, /obj/machinery/door/window)
+
+	for(var/dir in cardinal) // North, South, East, West
+		var/obj/structure/obstacle = locate(/obj/structure, get_step(src, dir))
+		if(is_type_in_list(obstacle, valid_obstacles))
+			obstacle.attack_animal(src)
+
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/UnlockBlastDoors(var/target_id_tag, var/msg_to_send)
+	var/unlocked_something = 0
+	for (var/obj/machinery/door/poddoor/P in world)
+		if (P.density && P.id_tag == target_id_tag && P.z == z)
+			P.open()
+			unlocked_something = 1
+	if (unlocked_something)
+		for(var/mob/living/carbon/human/H in player_list)
+			if (H.z != z)
+				continue
+			to_chat(H,"<span class='notice'>----------</span>")
+			to_chat(H,"<span class='notice'>[msg_to_send]</span>")
+			to_chat(H,"<span class='notice'>----------</span>")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoHiveSense()
+	var/hsline = ""
+	to_chat(src, "Your Brood: ")
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in mob_list)
```

Comment:
is there any reason you check for things in the `mob_list` then check for if they're alive? There's a `living_mob_list` for stuff like this.